llms/mlx_lm/tuner/utils.py

```diff
@@ -130,7 +132,15 @@ def to_lora(layer):
 
     for l in model.layers[num_layers - num_lora_layers :]:
         lora_layers = [(k, to_lora(m)) for k, m in l.named_modules() if k in keys]
-        l.update_modules(tree_unflatten(lora_layers))
+        if lora_layers:
+            l.update_modules(tree_unflatten(lora_layers))
+
+    non_layer_modules = [
+        (k, m) for k, m in model.named_modules() if "model.layers" not in k
+    ]
```

Comment:
This is a bug I think. The model keys are not typically `model.layers.*`. `model.layers` is a property for easy access to the models hidden layers / blocks.

Maybe the keys for the embedding and linear layers should be the full path to make this simpler? Not sure.. maybe there is a cleaner way.

Reply:
Sorry, I'm not quite sure I understand. Here I'm trying to gather all `named_modules()` that are not in in the hidden layers`"model.layers" not in k`. So `non_layer_modules` should be all modules - hidden layer modules. So now the `lm_head` and `embeddings` can be targeted from the set of modules that are not the hidden layer modules.

Reply:
Sorry for the delay. What I mean to say is `model.layers` is not always the path to the layers. Sometimes it's like `model.blocks` or something else. So `non_layer_modules` will not always be "all modules - hidden layer modules".

Reply:
If the keys for the non layer modules are always the full path then the above line is not really necessary is it?

Reply:
Great point about it being unnecessary. Will update.

Reply:
@awni updated and test still passing.